### PR TITLE
[Behat][Admin] Create delete admin feature

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | #5797 
| License         | MIT

### CODE REVIEW

File: features/user/managing_administrators/deleting_administrators.feature [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        And there should be 1 administrators in the list
+        And there should be no "mr.banana@example.com" administrator anymore
+
+    @ui
+    Scenario: The administrator account of currently logged in user cannot be deleted
+        Given I want to browse administrators
+        When I delete administrator with email "ted@example.com"
+        Then I should be notified that it cannot be deleted
+        And there should be 2 administrators in the list
+        And there should still be only one administrator with email "ted@example.com"
```
Comment (GSadee) [inline features/user/managing_administrators/deleting_administrators.feature]: Missing blank line at the end of file

File: features/user/managing_administrators/administrator_unique_email_and_name_validation.feature [hunk 2]
```diff
@@ -14,7 +14,6 @@ Feature: Administrator unique email and name validation
         When I specify its email as "ted@example.com"
         And I try to add it
         Then I should be notified that email must be unique
-        And I want to see all administrators in store
```
Comment (GSadee) [inline features/user/managing_administrators/administrator_unique_email_and_name_validation.feature]: ... in **the** store ?

File: features/user/managing_administrators/deleting_administrators.feature [hunk 3]
```diff
@@ -0,0 +1,26 @@
+@managing_administrators
+Feature: Deleting an administrator
+    In order to delete administrator account
```
Comment (GSadee) [inline features/user/managing_administrators/deleting_administrators.feature]: ... delete **an** administrator account ?

File: features/user/managing_administrators/deleting_administrators.feature [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        Given there is an administrator "mr.banana@example.com"
+        And there is also an administrator "ted@example.com"
+        And I am logged in as "ted@example.com" administrator
+
+    @ui
+    Scenario: Deleting an administrator
+        Given I want to browse administrators
+        When I delete administrator with email "mr.banana@example.com"
+        Then I should be notified that it has been successfully deleted
+        And there should be 1 administrators in the list
```
Comment (GSadee) [inline features/user/managing_administrators/deleting_administrators.feature]: ... should be **a single administrator** in the list

File: features/user/managing_administrators/deleting_administrators.feature [hunk 5]
```diff
@@ -0,0 +1,26 @@
+@managing_administrators
+Feature: Deleting an administrator
+    In order to delete administrator account
+    As an Administrator
+    I want to be able to delete the administrator
```
Comment (lchrusciel) [inline features/user/managing_administrators/deleting_administrators.feature]: delete the administaror? He won't be happy about that 💀 

`I want to be able to delete an other administrators account`?

File: features/user/managing_administrators/deleting_administrators.feature [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        And there is also an administrator "ted@example.com"
+        And I am logged in as "ted@example.com" administrator
+
+    @ui
+    Scenario: Deleting an administrator
+        Given I want to browse administrators
+        When I delete administrator with email "mr.banana@example.com"
+        Then I should be notified that it has been successfully deleted
+        And there should be 1 administrators in the list
+        And there should be no "mr.banana@example.com" administrator anymore
```
Comment (lchrusciel) [inline features/user/managing_administrators/deleting_administrators.feature]: 💀

File: features/user/managing_administrators/deleting_administrators.feature (hunk 1 above)
Comment (lchrusciel) [inline features/user/managing_administrators/deleting_administrators.feature]: How many of them should be here? You have not try to add a new one.

`And there should still be a "ted@example.com" administrator in registry` or `And my account should not be deleted`
But the second one would require additional transformer.

File: src/Sylius/Behat/Context/Ui/Admin/ManagingAdministratorsContext.php [hunk 7]
```diff
@@ -154,12 +164,23 @@ public function iSaveMyChanges()
...
+    public function iDeleteAdministratorWithEmail($email)
+    {
+        $this->indexPage->deleteResourceOnPage(['email' => $email]);
+    }
+
+    /**
      * @Then the administrator :email should appear in the store
      * @Then I should see the administrator :email in the list
      * @Then there should still be only one administrator with email :email
+     * @Then there should still be administrator with email :email
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingAdministratorsContext.php]: with **an** email

File: src/Sylius/Bundle/UserBundle/DependencyInjection/SyliusUserExtension.php [hunk 8]
```diff
@@ -90,6 +91,7 @@ private function createServices(array $resources, $driver, ContainerBuilder $con
             $this->createProviders($userType, $config['user']['classes']['model'], $container);
             $this->createFormTypes($userType, $config['user'], $container);
             $this->createAddUserTypeFromSubscribers($userType, $container);
+            $this->createUserDeleteListeners($userType, $container);
```
Comment (lchrusciel) [inline src/Sylius/Bundle/UserBundle/DependencyInjection/SyliusUserExtension.php]: :(

Comment (Arminek) [inline src/Sylius/Bundle/UserBundle/DependencyInjection/SyliusUserExtension.php]: :(

File: features/user/managing_administrators/deleting_administrators.feature (hunk 3 above)
Comment (michalmarcinkowski) [inline features/user/managing_administrators/deleting_administrators.feature]: How about `In order to get rid of deprecated administrators`?

File: features/user/managing_administrators/deleting_administrators.feature (hunk 5 above)
Comment (michalmarcinkowski) [inline features/user/managing_administrators/deleting_administrators.feature]: `I want to be able to delete other administrator account`

**Comment by pjedrzejewski** [comment]
Thank you Arek! :)

**Pull request merged**